src/State/CreateProvider.php

```diff
@@ -31,35 +30,34 @@
  * @author Antoine Bluchet <soyuka@gmail.com>
  *
  * @experimental
+ * @internal
  */
 final class CreateProvider implements ProviderInterface
 {
-    public function __construct(private ProviderInterface $decorated, private ?PropertyAccessorInterface $propertyAccessor = null)
-    {
+    public function __construct(
+        private ProviderInterface $provider,
+        private ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory,
+        private ?PropertyAccessorInterface $propertyAccessor = null,
+    ) {
         $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
     }
 
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): ?object
     {
         if (!$uriVariables || !$operation instanceof HttpOperation || null !== $operation->getController()) {
-            return $this->decorated->provide($operation, $uriVariables, $context);
+            return $this->provider->provide($operation, $uriVariables, $context);
         }
 
         $operationUriVariables = $operation->getUriVariables();
         $relationClass = current($operationUriVariables)->getFromClass();
         $key = key($operationUriVariables);
-        $relationUriVariables = [];
-
-        foreach ($operationUriVariables as $parameterName => $value) {
-            if ($key === $parameterName) {
-                $relationUriVariables['id'] = new Link(identifiers: $value->getIdentifiers(), fromClass: $value->getFromClass(), parameterName: $key);
-                continue;
-            }
 
-            $relationUriVariables[$parameterName] = $value;
+        try {
+            $parentOperation = $this->resourceMetadataCollectionFactory->create($relationClass)->getOperation($context['parent_uri_template'] ?? null);
```

Comment:
we should use the `operationMetadataFactory` instead

Reply:
Not sure what you mean here...

Reply:
oh yeah right now I understand better what this does, and no more weird uri variables guess...